dnstable_convert.c

```diff
@@ -1017,6 +1017,26 @@ init_mtbl(mtbl_compression_type compression, int level, size_t dns_block_size,
 	mtbl_writer_options_destroy(&wopt);
 }
 
+static bool
+parse_long(const char *str, long int *val)
+{
+	char *endptr;
+
+	errno = 0;
+	*val = strtol(str, &endptr, 0);
+
+	if ((errno == ERANGE && (*val == LONG_MAX || *val == LONG_MIN))
+	    || (errno != 0 && *val == 0))
+	{
+		return false;
+	}
+
+	if (endptr == str || *endptr != '\0')
```

Comment:
I think the latter half of the comparison suffices... `endptr == str` is really just a sub-case of `*endptr != '\0'`

Could even become `return (*endptr != '\0');`

Reply:
Good point; fixed!